### ISSUE TITLE
Wild Pyromancer + Cat Trick

### ISSUE THREAD

**Issue by DominusMaximus** [demilich1/metastone#240]
- It's clear if you play Wild Pyromancer and then Equality while your opponent has a Cat Trick, the 4/2 that is summoned won't be affected by Equality. My question is, because both cards are triggering off of the resolution of the same spell, will the 4/2 take the 1 damage from the Pyromancer trigger _only_ if the Cat Trick was played before the Pyromancer? Over at http://hearthstone.gamepedia.com/Triggered_effect it says:

"When multiple triggered effects activate at once, such as at the end of a player's turn, the effects will activate in the order in which the minions were summoned, with the minions summoned first activating their triggered effects first."

And then at http://hearthstone.gamepedia.com/Secret it says:

"When Secrets and triggered effects are activated from the same trigger, the effects will activate in the order in which they were played."

In Metastone, regardless of the order of play, the 4/2 will always take the 1 damage.

**Comment by webadict** [comment]
According to the cards, Cat Trick and Wild Pyro should be on the same trigger, so it should only matter on order of play. I'll do a quick check, though.

**Comment by webadict** [comment]
I think I know what the issue is...

**Comment by webadict** [comment]
Aha, I have found the issue, which I'm sure was a result of some old bugs that were happening. Triggers were associated with layers: A default layer, and a secret layer. This TYPICALLY doesn't cause any issues...

Well, that's not true. It actually caused several issues that I wasn't even paying attention to.

So, I removed that. The biggest issue this looks like it was solving was things like Mana Wyrm/Violet Teacher/etc. from triggering, but [Hearthstone's wiki](http://hearthstone.gamepedia.com/Counterspell) states otherwise:

> Any effects which activate from the spell being cast will still be triggered. Examples include Mana Wyrm, Questing Adventurer, Unbound Elemental and rogue combo effects on other cards.[2]

So, this actually fixes several bugs that nobody realized were bugs. Good catch.
